packages/dds/tree/src/test/feature-libraries/editable-tree/utilities.spec.ts

```diff
@@ -92,10 +93,22 @@ describe("editable-tree utilities", () => {
     });
 
     it("get array-like keys", () => {
-        assert.deepEqual(getArrayOwnKeys(1), Object.getOwnPropertyNames([""]));
-        assert.deepEqual(getArrayOwnKeys(0), Object.getOwnPropertyNames([]));
-        // TODO: make sure "length" is not configurable/enumerable when implementing proxy for arrayed fields
-        assert.deepEqual(getArrayOwnKeys(1), [...Object.keys([""]), "length"]);
-        assert.deepEqual(getArrayOwnKeys(0), [...Object.keys([]), "length"]);
+        assert.deepEqual(getOwnArrayKeys(1), Object.getOwnPropertyNames([""]));
+        assert.deepEqual(getOwnArrayKeys(0), Object.getOwnPropertyNames([]));
+        assert.deepEqual(getOwnArrayKeys(1), [...Object.keys([""]), "length"]);
+        assert.deepEqual(getOwnArrayKeys(0), [...Object.keys([]), "length"]);
+    });
+
+    it("key is a valid array index", () => {
+        assert.equal(keyIsValidIndex(0, 1), true);
+        assert.equal(keyIsValidIndex(0, 0), false);
+        assert.equal(keyIsValidIndex("0", 1), true);
+        assert.equal(keyIsValidIndex("0", 0), false);
```

Comment:
I think this is a case we should cover:
```suggestion
        assert.equal(keyIsValidIndex("0", 0), false);
        assert.equal(keyIsValidIndex("0.0", 1), false);
```

Maybe also cases for NaN and Infinity since they are strange.